### PR TITLE
tedge-apt-plugin update list missing keep config options

### PR DESCRIPTION
**Pull request by rina23q**
## Proposed changes

Refactor tedge-apt-plugin to move logic to a common area. It solves the linked tickets.

The systems tests were updated to improve coverage for the keep config options which was previously erroneously passing due to using manual `tedge-apt-plugin install` commands rather than installing packages from the cloud which uses a different code path to install packages.

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue

* https://github.com/thin-edge/thin-edge.io/issues/2669
* https://github.com/thin-edge/thin-edge.io/issues/2671
* https://github.com/thin-edge/thin-edge.io/issues/2670

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by reubenmiller — APPROVED** [review]
From an apt command point of view, I can confirm that the commands and test is good.

Though wait for a Rust review before merging.

**Review by didier-wenzek — APPROVED** [review]
Approved. Nicely done. This refactoring will help avoiding the same kind of issues.

That said, this PR description is not correct: it doesn't address https://github.com/thin-edge/thin-edge.io/issues/2666

**Comment by reubenmiller** [comment]
> Approved. Nicely done. This refactoring will help avoiding the same kind of issues.
> 
> That said, this PR description is not correct: it doesn't address #2666

Sorry that was my fault, I linked the wrong ticket...updating now.

**Pull request merged**
